src/Components/Server/src/ComponentHub.cs

```diff
@@ -123,7 +131,22 @@ public async Task<bool> ConnectCircuit(string circuitId)
         /// </summary>
         public void BeginInvokeDotNetFromJS(string callId, string assemblyName, string methodIdentifier, long dotNetObjectId, string argsJson)
         {
-            EnsureCircuitHost().BeginInvokeDotNetFromJS(callId, assemblyName, methodIdentifier, dotNetObjectId, argsJson);
+            if (assemblyName == "Microsoft.JSInterop" && methodIdentifier == "DotNetDispatcher.EndInvoke")
+            {
+                return;
```

Comment:
Seems like instead of having this check here, we should be removing the `[JSInvokable]` overload as per the comment at https://github.com/aspnet/Extensions/pull/2043#discussion_r303813358

Reply:
I'm giving that a try.

Reply:
I got rid of this by eliminating [JSInvokable] from EndInvoke

Reply:
I added a platform method on mono to replace the invocation

Reply:
Just to check, do you mean you put it on `Platform.ts` or on `Environment.ts`?

It should be on `Environment.ts` since "platform" refers to a .NET runtime (e.g., how to load assemblies and invoke .NET methods on them), whereas "environment" is a broader concept containing both the .NET runtime platform *and* any other environment-specific variations.

Currently `Environment.ts` only exports `platform`, so this is the perfect use case for adding another export there.

Reply:
I mean https://github.com/aspnet/Extensions/pull/2043/files#diff-1098eb9b27383934318e3bcbd83daa41R36

Reply:
Oh, I see - great!

Reply:
https://github.com/aspnet/AspNetCore/pull/12182/files#diff-dafa3a12ac230b86824394e3af774677R327

Reply:
done